RecoBTag/ONNXRuntime/plugins/DeepCombinedONNXJetTagsProducer.cc

```diff
@@ -0,0 +1,345 @@
+      output_tags.emplace_back(std::make_unique<JetTagCollection>(ref2prod));
+    }
+
+    // convert inputs
+    unsigned int inputs_done_count = 0;
+    for (unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n) {
+      const auto& jet_ref = (*tag_infos)[jet_n].jet();
+      if (jet_ref->pt() > min_jet_pt_ && std::fabs(jet_ref->eta()) < max_jet_eta_) {
+        const auto& taginfo = (*tag_infos)[jet_n];
+        make_inputs(inputs_done_count, taginfo);
+        inputs_done_count++;
+      }
+    }
+
+    // run prediction
+    assert(inputs_done_count == good_taginfo_count);
+    const auto outputs = globalCache()->run(input_names_, data_, {}, output_names_, good_taginfo_count)[0];
+    assert(outputs.size() == flav_names_.size() * good_taginfo_count);
+
+    // get the outputs
+    unsigned i_output = 0;
+    for (unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n) {
+      const auto& jet_ref = (*tag_infos)[jet_n].jet();
+      for (std::size_t flav_n = 0; flav_n < flav_names_.size(); flav_n++) {
+        if (jet_ref->pt() > min_jet_pt_ && std::fabs(jet_ref->eta()) < max_jet_eta_) {
+          (*(output_tags[flav_n]))[jet_ref] = outputs[i_output];
+          ++i_output;
+        } else
+          (*(output_tags[flav_n]))[jet_ref] = -2;
+      }
+    }
+  } else {
+    // create empty output collection
+    for (std::size_t i = 0; i < flav_names_.size(); i++) {
+      output_tags.emplace_back(std::make_unique<JetTagCollection>());
+    }
+  }
+
+  // put into the event
+  for (std::size_t flav_n = 0; flav_n < flav_names_.size(); ++flav_n) {
+    iEvent.put(std::move(output_tags[flav_n]), flav_names_[flav_n]);
+  }
+}
+
+void DeepCombinedONNXJetTagsProducer::make_inputs(unsigned i_jet, const reco::DeepFlavourTagInfo& taginfo) {
+  const auto& features = taginfo.features();
+  float* ptr = nullptr;
+  const float* start = nullptr;
+  unsigned offset = 0;
+
+  // jet and other global features
+  offset = i_jet * input_sizes_[kGlobal];
+  ptr = &data_[kGlobal][offset];
+  // jet variables
+  const auto& jet_features = features.jet_features;
+  start = ptr;
+  *ptr = jet_features.pt;
+  *(++ptr) = jet_features.eta;
+  // number of elements in different collections
+  *(++ptr) = features.c_pf_features.size();
+  *(++ptr) = features.n_pf_features.size();
+  *(++ptr) = features.sv_features.size();
+  *(++ptr) = features.npv;
+  // variables from ShallowTagInfo
+  const auto& tag_info_features = features.tag_info_features;
+  *(++ptr) = tag_info_features.trackSumJetEtRatio;
+  *(++ptr) = tag_info_features.trackSumJetDeltaR;
+  *(++ptr) = tag_info_features.vertexCategory;
+  *(++ptr) = tag_info_features.trackSip2dValAboveCharm;
+  *(++ptr) = tag_info_features.trackSip2dSigAboveCharm;
+  *(++ptr) = tag_info_features.trackSip3dValAboveCharm;
+  *(++ptr) = tag_info_features.trackSip3dSigAboveCharm;
+  *(++ptr) = tag_info_features.jetNSelectedTracks;
+  *(++ptr) = tag_info_features.jetNTracksEtaRel;
```

Comment:
this could also be a separate function in tensor_filler.cc, later to be integrated in DeepFlavour